internal/tofu/node_output.go

```diff
@@ -615,7 +615,7 @@ func (n *NodeApplyableOutput) setValue(state *states.SyncState, changes *plans.C
 
 	// non-root outputs need to keep sensitive marks for evaluation, but are
 	// not serialized.
-	if n.Addr.Module.IsRoot() {
+	if n.Addr.Module.IsRoot() && marks.Contains(val, marks.Sensitive) {
```

Comment:
This is actually irrelevant for the deprecation marks. However, we don't need to remove sensitive marks if there are none.